src/cnxml/document.js

```diff
@@ -133,18 +131,40 @@ export const FIGURE = block(
  * Process data for images.
  */
 function de_image(el) {
+    const data = {
+        mime: el.getAttribute('mime-type'),
+        src: el.getAttribute('src'),
+    }
+    if (el.hasAttribute('for')) {
+        data.for = el.getAttribute('for')
+    }
+    if (el.hasAttribute('height')) {
+        data.height = Number(el.getAttribute('height'))
+    }
+    if (el.hasAttribute('longdesc')) {
+        data.longdesc = el.getAttribute('longdesc')
+    }
+    if (el.hasAttribute('thumbnail')) {
+        data.thumbnail = el.getAttribute('thumbnail')
+    }
+    if (el.hasAttribute('width')) {
+        data.width = Number(el.getAttribute('width'))
+    }
```

Comment:
I'm not convinced we want to support all those attributes. I'm not sure whether the modern online/pdf pipeline even makes use of width, height, or thumbnail. As for longdesc, I have never seen it used in OpenStax's books and I don't know how it's supposed to be used / how it works. A similar [HTML property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/longDesc) is deprecated.

Reply:
I think `width` and `height` may be useful even if they are not used by OS.
About `thumbnail` and `longdesc` - I've added those just because they exists in `cnxml` and I'm not sure if we should remove them if they were present when loading document

Reply:
We shouldn't support something just because it may be useful. Also, the seem too much like formatting to me, and we explicitly don't support formatting.

As for removing attributes; we already don't guarantee that the editor will preserve anything, and in fact it often modifies documents, so I don't think that's a problem.

Reply:
> we already don't guarantee that the editor will preserve anything

I know, but I don't think this is good approach. I would like it to be more like: preserve anything which we can and allow editing things which we want to be edited

Let me know which attributes you would like to support from:
https://legacy.cnx.org/eip-help/image
https://legacy.cnx.org/eip-help/audio
https://legacy.cnx.org/eip-help/video

Reply:
Decision: as little as possible

Reply:
According to: https://katalysteducation.slack.com/archives/GCSUE8C01/p1576223516003500
support those tags which can be used in cnxml and disable those which we does not want to support in front end